src/fields/field_fe.cc

```diff
@@ -290,11 +310,132 @@ void FieldFE<spacedim, Value>::value_list (const std::vector< Point >  &point_li
 
 
 template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::init_from_input(const Input::Record &rec, const struct FieldAlgoBaseInitData& init_data) {
+	this->init_unit_conversion_coefficient(rec, init_data);
+	flags_ = init_data.flags_;
+
+
+	// read mesh, create tree
+    {
+       reader_file_ = FilePath( rec.val<FilePath>("mesh_data_file") );
+       auto reader_data = ReaderInstance::get_instance(reader_file_);
+       source_mesh_ = reader_data.mesh_;
+       source_mesh_->get_bih_tree(); // only create BIH tree
+	   // no call to mesh->setup_topology, we need only elements, no connectivity
+    }
+
+	field_name_ = rec.val<std::string>("field_name");
+}
+
+
+
+template <int spacedim, class Value>
+void FieldFE<spacedim, Value>::set_mesh(const Mesh *mesh, bool boundary_domain) {
```

Comment:
Field factory creates FieldFE, but for this case (piezo_head_p0) FieldFE is apparently encapsulated in FieldAddPotential. Is it possible?
I suggest to implement FieldAddPotential::set_mesh method (similarly as set_time).